proc/proc.go

```diff
@@ -109,48 +112,111 @@ func GetKernelModules(modulesPath string,
 		Size:    int(etext.Address - stext.Address),
 	})
 
-	atLeastOneValidAddress := false
-	count := 0
+	content, err := io.ReadAll(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read modules file: %v", err)
+	}
 
-	var scanner = bufio.NewScanner(file)
-	for scanner.Scan() {
-		var size, address uint64
-		var refcount int64
-		var name, dependencies, state string
+	modules, err := parseKernelModules(content)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel modules: %v", err)
+	}
+
+	for _, kmod := range modules {
+		symmap.Add(libpf.Symbol{
+			Name:    libpf.SymbolName(kmod.name),
+			Address: libpf.SymbolValue(kmod.address),
+			Size:    kmod.size,
+		})
+	}
+
+	symmap.Finalize()
 
+	return &symmap, nil
+}
+
+func parseKernelModules(content []byte) ([]kernelModule, error) {
+	var (
+		modules                []kernelModule
+		scanner                = bufio.NewScanner(bytes.NewReader(content))
+		atLeastOneValidAddress = false
+		count                  = 0
+	)
+
+	for scanner.Scan() {
 		line := scanner.Text()
 
 		count++
 
-		nFields, err := fmt.Sscanf(line, "%s %d %d %s %s 0x%x",
-			&name, &size, &refcount, &dependencies, &state, &address)
+		kmod, err := parseKernelModuleLine(line)
 		if err != nil {
-			log.Warnf("err parsing line in modules: '%s'", err)
-			continue
+			return nil, fmt.Errorf("failed to parse kernel module line: %v", err)
 		}
-		if nFields < 6 {
-			log.Warnf("unexpected line in modules: '%s'", line)
-			continue
-		}
-		if address == 0 {
+		if kmod.address == 0 {
 			continue
 		}
 		atLeastOneValidAddress = true
 
-		symmap.Add(libpf.Symbol{
-			Name:    libpf.SymbolName(name),
-			Address: libpf.SymbolValue(address),
-			Size:    int(size),
-		})
+		modules = append(modules, kmod)
 	}
 
 	if count > 0 && !atLeastOneValidAddress {
 		return nil, errors.New("addresses from all modules is zero - check process permissions")
 	}
 
-	symmap.Finalize()
+	return modules, nil
+}
 
-	return &symmap, nil
+type kernelModule struct {
+	name    string
+	size    int
```

Comment:
I have a slight preference to being explicit about sizes (e.g. int64 if this can indeed be bigger than 2 GiB)

Reply:
I only went with int because the rest of the codebase used ints where this ends up being used, but let me see how much work it would be to change this.